CIP-0095/README.md

```diff
@@ -0,0 +1,771 @@
+containing vote delegation certificates. The wallet should articulate this
+request from client application in a explicit and highly informative way. Users
+must be shown the target of the delegation (DRepID or a predefined DRep
+identifier) and must be informed which of their stake keys are being used. For
+the case of multiple delegations at once, the user may only be asked for
+permission to authorize all at once, but this can depend on wallet's
+implementation.
+
+If user grants permission, each transaction must be signed by the secret key of
+the provided public stake key, with the signature and key to be added to the
+transaction witness set before submission.
+
+By allowing clients to supply the stake key we are placing the burden of
+"multi-governance-delegation" management onto the client, reducing the
+complexity for wallet implementations. By forcing wallets to inspect these
+certificates it allows the user to catch malicious client applications
+attempting to insert their own delegation targets.
+
+##### Errors
+
+One `TxSignError` should be returned if there is a signature error with any of
+the certificates.
+
+##### Returns
+
+This returns an array of `SubmittedTransaction` objects which contain the
+details of the submitted delegation certificates, for the client to confirm. The
+returned `txHash`s can be used by the client to track the status of the
+transactions containing the certificates on Cardano.
+
+#### `api.submitDRepRegistration(tx: UnsignedTransaction): Promise<SubmittedTransaction>`
+
+Errors: [APIError](#extended-apierror), [`TxSignError`](#extended-txsignerror)
+
+This endpoint requests the wallet to inspect, sign and submit a transaction
+containing a DRep registration certificate. The wallet should articulate this
+request from client application in a explicit and highly informative way. Users
+should be made aware of the type of certificate, associated DRepID, metadata
+anchor and deposit amount.
+
+If user grants permission, the transaction must be signed by the secret DRep key
+as described in [DRep Key](#drep-key), with the signature and key to be added to
+the transaction witness set before submission.
+
+By allowing clients to choose UTxOs we are placing the burden of managing a
+user's DRep registration deposit on the application. By forcing wallets to
+inspect these certificates it allows the user to catch malicious client
+applications attempting to insert their own data into the certificate.
+
+##### Errors
+
+One `TxSignError` should be returned if there is a signature error with any of
+the certificates.
+
+##### Returns
+
+This returns a `SubmittedTransaction` object which contains all the details of
+the submitted registration certificate, for the client to confirm. The returned
+`txHash` can be used by the client to track the status of the transaction
+containing the certificate on Cardano.
+
+#### `api.submitDRepRetirement(tx: UnsignedTransaction): Promise<SubmittedTransaction>`
```

Comment:
If these are regular cardano transactions, couldn't the DApp use the basic `signTransaction()` call? The logic required to verify that the unsigned transaction has the required structure should be not significantly different from any existing parsing logic of the wallets to show a user-friendly summary of the transaction signed via the CIP-30 call.

The Same applies for `submitDRepRegistration` call.

Reply:
Also, I am trying to avoid issues with mismatch between CIP30 implementations - its unknown how existing CIP30 endpoints would be able to interact with CIP1694/Conway Ledger era items. I discuss this in [CIP-30 Reuse](https://github.com/Ryun1/CIPs/blob/governance-wallet-connector/CIP-0095/README.md#cip-30-reuse).

Reply:
I think the answer to my question above: https://github.com/cardano-foundation/CIPs/pull/509/files#r1260839044 would answer also this question

Reply:
See answer [comment](https://github.com/cardano-foundation/CIPs/pull/509#discussion_r1260873125).